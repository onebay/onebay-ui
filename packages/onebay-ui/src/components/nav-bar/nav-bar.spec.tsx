import { mount } from '@vue/test-utils'
import NavBar from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(NavBar, {
    components: {},
    slots,
    props: { ...values },
  })
}

describe('NavBar Snap', () => {
  it('render initial NavBar', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render NavBar -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render NavBar -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render NavBar -- props fixed', () => {
    const wrapper = factory({ fixed: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render NavBar -- props color', () => {
    const wrapper = factory({ color: '#fff' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render NavBar -- props leftContent', () => {
    const wrapper = factory({
      color: '#fff',
      leftContent: <div class="ob-icon ob-icon-chevron-left"></div>
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render NavBar -- props title', () => {
    const wrapper = mount({
      render() {
        return (
          <NavBar>
            <span>test</span>
          </NavBar>
        )
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('NavBar Behavior ', () => {
  it('NavBar onClickLeftIcon', () => {
    const onClickLeftIcon = jest.fn()
    const wrapper = factory({
      leftContent: <div class="ob-icon ob-icon-chevron-left"></div>
    })
    wrapper.find('.ob-nav-bar__left-view div').trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('NavBar click right first icon', () => {
    const wrapper = factory({
      rightContent: <>
        <div class="ob-icon ob-icon-bullet-list"></div>
        <div class="ob-icon ob-icon-user"></div>
      </>
    })
    wrapper.find('.ob-nav-bar__right-view div:first-child').trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('NavBar click right second icon', () => {
    const wrapper = factory({
      rightContent: <>
        <div class="ob-icon ob-icon-bullet-list"></div>
        <div class="ob-icon ob-icon-user"></div>
      </>
    })
    wrapper.find('.ob-nav-bar__right-view div:last-child').trigger('click')
    expect(wrapper.element).toMatchSnapshot()
  })
})
