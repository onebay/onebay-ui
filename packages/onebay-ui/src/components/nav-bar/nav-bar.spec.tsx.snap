// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Behavior  NavBar click right first icon 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    
    <div
      class="ob-icon ob-icon-bullet-list"
    />
    <div
      class="ob-icon ob-icon-user"
    />
    
  </div>
</div>
`;

exports[`NavBar Behavior  NavBar click right second icon 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    
    <div
      class="ob-icon ob-icon-bullet-list"
    />
    <div
      class="ob-icon ob-icon-user"
    />
    
  </div>
</div>
`;

exports[`NavBar Behavior  NavBar onClickLeftIcon 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <div
      class="ob-icon ob-icon-chevron-left"
    />
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render NavBar -- props className 1`] = `
<div
  class="ob-nav-bar test"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render NavBar -- props color 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
    style="color: rgb(255, 255, 255);"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render NavBar -- props customStyle 1`] = `
<div
  class="ob-nav-bar"
  style="color: red;"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render NavBar -- props fixed 1`] = `
<div
  class="ob-nav-bar ob-nav-bar--fixed"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render NavBar -- props leftContent 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
    style="color: rgb(255, 255, 255);"
  >
    <div
      class="ob-icon ob-icon-chevron-left"
    />
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render NavBar -- props title 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    <span>
      test
    </span>
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;

exports[`NavBar Snap render initial NavBar 1`] = `
<div
  class="ob-nav-bar"
>
  <div
    class="ob-nav-bar__left-view"
  >
    <!---->
  </div>
  <div
    class="ob-nav-bar__title"
  >
    
    
  </div>
  <div
    class="ob-nav-bar__right-view"
  >
    <!---->
  </div>
</div>
`;
